sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala

```diff
@@ -91,4 +92,16 @@ class PruneFileSourcePartitionsSuite extends QueryTest with SQLTestUtils with Te
       assert(size2 < tableStats.get.sizeInBytes)
     }
   }
+
+  test("SPARK-26576 Broadcast hint not applied to partitioned table") {
+    withTable("tbl") {
+      spark.range(10).selectExpr("id", "id % 3 as p").write.partitionBy("p").saveAsTable("tbl")
+
+      val df = spark.table("tbl")
+      val hints = df.join(broadcast(df), "p").queryExecution.optimizedPlan.collect {
```

Comment:
can we check the physical plan to make sure BroadcastJoin is picked?

Reply:
+1. And since master doesn't use ResolvedHint in optimization anymore, this issue doesn't exist in master, but figured out it wouldn't hurt to add a test in master too.

Reply:
Fine with porting at least the unit test to master.